File: Maximum_depth_of_binary_tree/solution_levelorderTraversal.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
public:
    int maxDepth(TreeNode *root) {
        if(root==NULL) return 0;
        int level = 0;
        queue<TreeNode*> cur_level,next_level;
        queue<TreeNode*> *cur, *next, *temp;
        cur = &cur_level;
        next = &next_level;
        cur_level.push(root);
        while(cur->size()!=0){
            if(cur->front()->left) next->push(cur->front()->left);
            if(cur->front()->right) next->push(cur->front()->right);
            cur->pop();
            if(cur->size()==0){
                temp = cur;
                cur = next;
                next = temp;
                level++;
            }
        }
        return level;
    }
};

File: Maximum_depth_of_binary_tree/solution_levelorderTraversal.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return an integer
    def maxDepth(self, root):
        if root is None:
            return 0
        level = 0    
        cur_level = []
        next_level = []
        cur_level.append(root)
        while len(cur_level) is not 0:
            if cur_level[0].left is not None:
                next_level.append(cur_level[0].left)
            if cur_level[0].right is not None:
                next_level.append(cur_level[0].right)
            del cur_level[0]
            if len(cur_level)==0:
                temp = cur_level
                cur_level = next_level
                next_level = temp
                level += 1
        return level
                
        
        
            
        